core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -1073,6 +1074,41 @@ object UniqueTriggerActivation {
                 }
             }
 
+            UniqueType.OneTimeRemoveResourcesFromTile -> {
+                if (tile == null) return null
+                if (tile.resource == null) return null
+                val resourceFilter = unique.params[0]
+                var tileResource = tile.tileResource
+                if (!tileResource.matchesFilter(resourceFilter)) return null
+                return {
+                    tile.resource = null
+                    tile.resourceAmount = 0
+                    true
+                }
+            }
+
+            UniqueType.OneTimeRemoveImprovementsFromTile -> {
+                if (tile == null) return null
+                val improvementFilter = unique.params[0]
+                val tileImprovement = tile.getTileImprovement()
+                if (tileImprovement == null) return null
+                if (!tileImprovement.matchesFilter(improvementFilter)) return null
+                return {
+                    // Don't remove the improvement if we're just removing the roads
+                    if (improvementFilter != "All Road") {
+                        tile.improvement = null
```

Comment:
tile.setImprovent(null) handles this case

Reply:
Amazing! Do we already have the Magnus unique from Civ6? I thought I had submitted a PR for it, but can't find the unique for it, maybe I only created it locally.